spark3-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ReplaceDataExec.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.connector.write.BatchWrite
+import org.apache.spark.sql.execution.SparkPlan
+
+case class ReplaceDataExec(batchWrite: BatchWrite, queryExec: SparkPlan) extends V2TableWriteExec {
+
+  // override child so that we can properly plan queryExec without executing the file filter
+  override def child: SparkPlan = queryExec
+
+  // TODO: doPrepare is NOT invoked by V2TableWriteExec
```

Comment:
I thought that `doPrepare` was called on all child nodes. So you can't add the `doPrepare` here, but it should work for any child. Is that not the case?

Reply:
I think the code below can be as simple as this:

```
  override lazy val query: SparkPlan = {
    queryExec.prepare()
    queryExec
  }
```

Is this what you mean, @rdblue?

Reply:
I thought that prepare should automatically be called, but it looks like our fix is to update `result` to call `prepare` and then `run`.

That would work here:

```
  def run(): ... {
    prepare()
    ...
  }
```

Calling it in `query` should work, too.

Reply:
Simplified.